c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -1631,7 +1631,11 @@ class THCCachingAllocator {
   }
 };
 
-THCCachingAllocator caching_allocator;
+
+THCCachingAllocator* get_caching_allocator() {
+  static auto r = new THCCachingAllocator();
```

Comment:
maybe include the information from the diff summary in a comment here (something like "leak to avoid LSAN failures; we rely on process cleanup anyway. if you are here because this is impacting you negatively, you can work around this by calling empty_cache before unloading libtorch.")